chapters/slang.adoc

```diff
@@ -0,0 +1,744 @@
+}
+----
+
+=== Step 3: Leverage Imports Instead of Includes
+
+Replace `#include` directives with Slang's import system:
+
+Before (HLSL):
+[source,hlsl]
+----
+#include "common.hlsl"
+#include "lighting.hlsl"
+float3 calculateLighting(...) {
+    // Use functions from included files
+}
+----
+
+After (Slang):
+[source,slang]
+----
+module Shaders.Fragment;
+import Shaders.Common;
+import Shaders.Lighting;
+export float3 calculateLighting(...) {
+    // Use functions from imported modules
+}
+----
+
+=== Step 4: Refactor Resource Bindings
+
+Update resource bindings to use Slang's parameter block system:
+
+Before (HLSL):
+[source,hlsl]
+----
+Texture2D albedoMap : register(t0);
+SamplerState samplerState : register(s0);
+cbuffer MaterialParams : register(b0) {
+    float4 baseColor;
+    float roughness;
+    float metallic;
+};
+----
+
+After (Slang):
+[source,slang]
+----
+struct MaterialResources {
+    Texture2D albedoMap;
+    SamplerState samplerState;
+    struct Params {
+        float4 baseColor;
+        float roughness;
+        float metallic;
+    } constants;
+};
+ParameterBlock<MaterialResources> material;
+// Usage
+float4 albedo = material.albedoMap.Sample(material.samplerState, uv);
+float roughness = material.constants.roughness;
+----
+
+=== Step 5: Introduce Generics and Interfaces
+
+Identify opportunities to use generics and interfaces for more flexible code:
+
+Before (HLSL):
+[source,hlsl]
+----
+float3 evaluateLambert(float3 albedo, float3 normal, float3 lightDir) {
+    return albedo * max(0, dot(normal, lightDir)) / 3.14159;
+}
+float3 evaluateGGX(float3 specColor, float roughness, float3 normal, float3 viewDir, float3 lightDir) {
+    // GGX implementation
+}
+float3 evaluateMaterial(MaterialType type, ...) {
+    switch(type) {
+        case MATERIAL_LAMBERT: return evaluateLambert(...);
+        case MATERIAL_GGX: return evaluateGGX(...);
+        default: return float3(0,0,0);
+    }
+}
+----
+
+After (Slang):
+[source,slang]
+----
+interface IBRDF {
+    float3 evaluate(float3 normal, float3 viewDir, float3 lightDir);
+}
+struct LambertBRDF : IBRDF {
+    float3 albedo;
+    float3 evaluate(float3 normal, float3 viewDir, float3 lightDir) {
+        return albedo * max(0, dot(normal, lightDir)) / 3.14159;
+    }
+}
+struct GGXBRDF : IBRDF {
+    float3 specColor;
+    float roughness;
+    float3 evaluate(float3 normal, float3 viewDir, float3 lightDir) {
+        // GGX implementation
+    }
+}
+float3 evaluateMaterial(IBRDF brdf, float3 normal, float3 viewDir, float3 lightDir) {
+    return brdf.evaluate(normal, viewDir, lightDir);
+}
+----
+
+=== Step 6: Implement Advanced Metaprogramming
+
+Use Slang's metaprogramming capabilities for more powerful shader generation:
+
+[source,slang]
+----
+// Define shader permutations using compile-time parameters
+[shader("vertex")]
+[CompileTimeConstant(name="USE_NORMAL_MAPPING", type="bool")]
+[CompileTimeConstant(name="LIGHT_COUNT", type="int")]
```

Comment:
If these `CompileTimeConstant` attributes were a thing at some point, they sure aren't anymore.
I'm pretty sure this example could be written like this instead:
`VSOutput vertexShader<bool UseNormalMapping, int LightCount>(VSInput input) {`.